src/modules/ie-objects/hooks/get-ie-objects-info.ts

```diff
@@ -16,11 +17,28 @@ export const useGetIeObjectInfo = (
 		enabled?: boolean;
 	} = {}
 ): UseQueryResult<IeObject | null> => {
-	return useQuery([QUERY_KEYS.getIeObjectsInfo, id], () => getIeObjectInfo(id), {
-		keepPreviousData: true,
-		enabled: true,
-		...options,
-	});
+	const router = useRouter();
+	return useQuery(
+		[QUERY_KEYS.getIeObjectsInfo, id],
+		async () => {
+			let newId;
+			if (id.length > 36) {
+				const v3Identifier = await IeObjectsService.schemaIdentifierLookup(id);
+				newId = v3Identifier.schemaIdentifierV3;
+				const url = new URL(window.location.href);
+				url.pathname = url.pathname.replace(id, newId);
+				await router.replace(url.toString());
```

Comment:
it might be better to move this code to the detail page
since this hook should only get some info from the api, and not have any other side effects

imagine we want to get an ie object somewhere else in the code, we might not want to do the redirect there.

something like:
```typescript
useEffect(() => {
	if (router.query.id.length > MIN_LENGTH_SCHEMA_IDENTIFIER_V2 && router.query.id !== mediaItem.schema_identifier) {
		const url = router.asPath;
		url= url.replace('/' + id + '/', '/' + newId + '/');
		await router.replace(url);
	}
}, [mediaItem]);
```